app/cdap/components/WranglerGrid/SelectColumnPanel/ColumnTable/index.tsx

```diff
@@ -0,0 +1,162 @@
+  height: 100%;
+`;
+
+const SelectColumnTable = styled(Table)`
+  display: flex;
+  flex-direction: column;
+`;
+
+const SelectColumnTableHead = styled(TableHead)`
+  height: 54px;
+`;
+
+const SelectColumnTableRow = styled(TableRow)`
+  font-weight: 400;
+  font-size: 16px;
+  line-height: 150%;
+  letter-spacing: 0.15px;
+  color: ${grey[700]};
+  display: grid;
+  grid-template-columns: 8% 45% 45%;
+  align-items: center;
+  height: 100%;
+`;
+
+const SelectColumnTableHeadCell = styled(TableCell)`
+  &.MuiTableCell-head {
+    padding: 0;
+    font-weight: 400;
+    font-size: 16px;
+    line-height: 150%;
+    letter-spacing: 0.15px;
+    color: ${grey[900]};
+    border-bottom: none !important;
+  }
+`;
+
+export default function({
+  columns,
+  transformationDataType,
+  onSingleSelection,
+  selectedColumns,
+  dataQualityValue,
+  isSingleSelection,
+  handleDisableCheckbox,
+  onMultipleSelection,
+  totalColumnCount,
+  setSelectedColumns,
+  transformationName,
+}: IColumnTableProps) {
+  const handleChange = () => {
+    if (
+      multipleColumnSelected?.filter(
+        (option) => option.value === transformationName && option.isMoreThanTwo === false
+      ).length > 0
+    ) {
+      if (selectedColumns.length) {
+        setSelectedColumns([]);
+      } else {
+        if (columns?.length > 2) {
+          setSelectedColumns(columns.slice(0, 2));
+        } else {
+          setSelectedColumns(columns);
+        }
+      }
+    } else {
+      if (selectedColumns?.length) {
+        setSelectedColumns([]);
+      } else {
+        setSelectedColumns(columns);
+      }
+    }
+  };
+
+  return (
+    <SelectColumnTableContainer data-testid="column-table-parent">
+      <SelectColumnTable aria-label="recipe steps table">
+        <Divider />
+        <SelectColumnTableHead>
+          <SelectColumnTableRow>
+            <SelectColumnTableHeadCell>
+              {multipleColumnSelected?.filter((option) => option.value === transformationName)
+                .length > 0 && (
+                <Checkbox
+                  color="primary"
+                  checked={selectedColumns?.length ? true : false}
+                  onClick={handleChange}
+                  indeterminate={selectedColumns?.length ? true : false}
+                  data-testid="column-table-check-box"
+                />
+              )}
+            </SelectColumnTableHeadCell>
+            <SelectColumnTableHeadCell data-testid="panel-columns">
+              {T.translate(`${ADD_TRANSFORMATION_PREFIX}.columns`)}
+              {` (${columns.length})`}
+            </SelectColumnTableHeadCell>
+            <SelectColumnTableHeadCell data-testid="panel-values">
+              {T.translate(`${ADD_TRANSFORMATION_PREFIX}.nullValues`)}
+            </SelectColumnTableHeadCell>
+          </SelectColumnTableRow>
+        </SelectColumnTableHead>
+        <Divider />
+        <TableBody>
+          {columns.map((eachColumn, columnIndex) =>
```

Comment:
It would be cleaner to hoist the column logic up out of the render, e.g.
```
const columnsToDisplay = transformationDataType?.includes('all') ? columns : columns.filter(...);
```
since the rendering blocks are the same.

Reply:
Written it with above syntax